### PR TITLE
solve issue related to validate ssh with tunnels

### LINKED ISSUE #37: [SSH Tunnelin: Validate SSH Not Required for HTTP connections
### What happened?

Hi, I was testing this package with SAS Viya and found this issue with df2sd.

withing the df2sd function validate_ssh_with_tunnel(sas_session)  is called without checking the connection type. This fails for HTTP / HTTPS connections. 

I was able to bypass this and directly call df2sd using the session object returned from saspy.

What did not work:

df2sd(mtcars, "mt")

Error in validate_ssh_with_tunnel(sas_session) : 
  SAS session through ssh must use tunnels to transfer datasets!

What worked:
 sas_session=get_sas_session()
 sas_session$df2sd(mtcars,"mt")

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by clarkliming**
close #37

### CODE REVIEW

**Comment by clarkliming** [comment]
hi @samiulhq sorry for the late reply. Hope this PR fix the issue you meet.

File: R/utils.R [hunk 1]
```diff
@@ -34,9 +34,11 @@ install_saspy <- function(method = "auto", conda = "auto") {
 #'
 #' @keywords internal
 validate_ssh_with_tunnel <- function(session, msg = "SAS session through ssh must use tunnels to transfer datasets!") {
-  cfgname <- session$sascfg$SAScfg$SAS_config_names[1]
+  cfgname <- session$sascfg$name
+  is_ssh <- identical(session$sascfg$mode, "ssh")
```
Comment (tomweber-sas) [inline R/utils.R:38]: I think identical() will only be TRUE if you match case. So use "SSH" for the literal, as the mode attribute is upper cased in saspy. 
Out of curiosity, why require tunneling with the SSH access method? It seems that any connection that's established via SASPy works, so why require this in the first place? I don't have an R install that works with this, so I haven't been able to try it out to see if I can answer that question myself, but it seems that this shouldn't care how saspy is connected; just that it is.
Thanks,
Tom

Comment (clarkliming) [reply R/utils.R:38]: thank you Tom. indeed this is upper case. the reason I add it is that a while ago I experienced issues transfering data for ssh without tunneling, but it seems it is working as expected now.
thank you so much!

**Review by tomweber-sas — APPROVED** [review]
Hey, I'm not expecting I can review these, such that it triggers the merge; says only reviewers w/ write permission, which I assume I'm not. But, adding this review for info anyway. This all looks good to me; I can't test it all out, but looking at the diffs, I don't see anything logically wrong. Looks like you're just allowing any valid connection, which is the right thing.
Don't know if this will trigger anything, but maybe helps with a reviewer who can!
Tom

**Comment by clarkliming** [comment]
> Hey, I'm not expecting I can review these, such that it triggers the merge; says only reviewers w/ write permission, which I assume I'm not. But, adding this review for info anyway. This all looks good to me; I can't test it all out, but looking at the diffs, I don't see anything logically wrong. Looks like you're just allowing any valid connection, which is the right thing. Don't know if this will trigger anything, but maybe helps with a reviewer who can! Tom

thank you Tom for your review!

**Review by danielinteractive — APPROVED** [review]
thanks @clarkliming !

**Pull request merged**